### PR TITLE
Dependency update 09/22

### PR DESCRIPTION
**Pull request by Rosomack**

### CODE REVIEW

File: orbit-core/src/commonMain/kotlin/org/orbitmvi/orbit/internal/RealContainer.kt [hunk 1]
```diff
@@ -71,9 +69,7 @@ public class RealContainer<STATE : Any, SIDE_EFFECT : Any>(
             internalStateFlow.value
         },
         reduce = { reducer ->
-            mutex.withLock {
-                internalStateFlow.value = reducer(internalStateFlow.value)
-            }
+            internalStateFlow.update(reducer)
```
Comment (Rosomack) [inline orbit-core/src/commonMain/kotlin/org/orbitmvi/orbit/internal/RealContainer.kt:72]: Notable change @mattmook, was able to get rid of mutex.

File: buildSrc/src/main/kotlin/DependencyManagement.kt [hunk 2]
```diff
@@ -20,29 +20,29 @@
...
-    const val safeargs = "2.3.5"
-    const val atomicfu = "0.17.0"
-    const val dokka = "1.6.10"
+    const val detektPlugin = "1.21.0"
+    const val safeargs = "2.5.1"
+    const val atomicfu = "0.18.3"
+    const val dokka = "1.7.10"
 
-    const val kotlin = "1.6.10"
+    const val kotlin = "1.7.10"
```
Comment (mattmook) [inline buildSrc/src/main/kotlin/DependencyManagement.kt:32]: coroutines out of sync with kotlin version?
latest mt version is 1.6.3-native-mt, but I see there's also 1.6.4 without native-mt version

File: orbit-core/src/commonMain/kotlin/org/orbitmvi/orbit/internal/RealContainer.kt (hunk 1 above)
Comment (mattmook) [reply orbit-core/src/commonMain/kotlin/org/orbitmvi/orbit/internal/RealContainer.kt:72]: yeah cool, was discussing this with someone not too long ago :)

File: buildSrc/src/main/kotlin/DependencyManagement.kt (hunk 2 above)
Comment (mattmook) [reply buildSrc/src/main/kotlin/DependencyManagement.kt:32]: ah further down we have upgraded to 1.6.4, just not here

Comment (Rosomack) [reply buildSrc/src/main/kotlin/DependencyManagement.kt:32]: according to the docs this should stay as-is https://kotlinlang.org/docs/multiplatform-mobile-concurrency-and-coroutines.html

**Review by mattmook — APPROVED** [review]

**Review by mattmook — DISMISSED** [review]

**Review by mattmook — APPROVED** [review]

**Review by mattmook — DISMISSED** [review]

**Review by mattmook — APPROVED** [review]

**Pull request merged**